vllm/model_executor/models/plamo2.py

```diff
@@ -0,0 +1,868 @@
+            input_buffers, **kwargs)
+
+    def get_seqlen_agnostic_capture_inputs(self, batch_size: int):
+        return self.mamba_cache.get_seqlen_agnostic_capture_inputs(batch_size)
+
+    def _get_mamba_cache_shape(
+            self) -> Tuple[Tuple[int, int], Tuple[int, int]]:
+        world_size = get_tensor_model_parallel_world_size()
+        hidden_size = (self.config.mamba_num_heads *
+                       self.config.hidden_size_per_head)
+        conv_state_shape = (
+            hidden_size // world_size,
+            self.config.mamba_d_conv - 1,
+        )
+        temporal_state_shape = (
+            hidden_size // world_size,
+            self.config.mamba_d_state,
+        )
+        return conv_state_shape, temporal_state_shape
+
+    def compute_logits(
+        self,
+        hidden_states: torch.Tensor,
+        sampling_metadata: SamplingMetadata,
+    ) -> Optional[torch.Tensor]:
+        logits = self.logits_processor(self.lm_head, hidden_states,
+                                       sampling_metadata)
+        return logits
+
+    def sample(
+        self,
+        logits: Optional[torch.Tensor],
+        sampling_metadata: SamplingMetadata,
+    ) -> Optional[SamplerOutput]:
+        next_tokens = self.sampler(logits, sampling_metadata)
+        return next_tokens
+
+    def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
+        params_dict = dict(self.named_parameters())
+        for name, loaded_weight in weights:
+
+            # Both tie_word_embeddings=True and lm_head.weight in the safetensor
+            # at the same time causes dict key access error.
+            if name == "lm_head.weight" and self.config.tie_word_embeddings:
+                assert "lm_head.weight" not in params_dict
+                continue
+
+            # Update the weight names to be compatible with the vllm version
+            # of the model.
+            # Do not change the order of the replacements.
+            replacements = {
+                # Skip PlamoDecoderLayers.
+                ".layers.layers": ".layers",
+                # Skip PlmoDecoderLayer.
+                ".mixer": "",
+                # Rename the final layernorm of the model.
+                "model.norm.weight": "model.final_layernorm.weight",
+
+                # Rename each mamba layer's components.
+                ".A_log": ".mamba.A",
+                ".B_norm_weight": ".mamba.b_layernorm.weight",
+                ".C_norm_weight": ".mamba.c_layernorm.weight",
+                ".dt_norm_weight": ".mamba.dt_layernorm.weight",
+                ".bcdt_proj.weight": ".mamba.x_proj.weight",
+                ".conv1d.weight": ".mamba.conv1d.weight",
+                ".in_proj.weight": ".mamba.in_proj.weight",
+                ".out_proj.weight": ".mamba.out_proj.weight",
+                ".D": ".mamba.D",
+                ".dt_bias": ".mamba.dt_bias",
+                ".dt_proj.weight": ".mamba.dt_proj.weight",
```

Comment:
I think it would be better to keep the model's member variable names consistent with the transformers implementation instead of adding this renaming step during `load_weights`

Reply:
Thanks for your suggestion. I've reverted some parts to the original class structure to minimize renaming.